### PR TITLE
Expand and clarify docs on persistent tasks

### PR DESCRIPTION
**Pull request by timholy**
While helping someone on slack I noticed these docs were a bit less clear than they could be. Here's my attempt at improving them. CC @NHDaly.

### CODE REVIEW

File: docs/src/persistent_tasks.md [hunk 1]
```diff
@@ -32,16 +43,31 @@ fails to precompile: `using PkgA` runs `PkgA.__init__()`, which
 leaves the `Timer` `Task` running, and that causes precompilation
 of `PkgB` to hang.
 
-## Example with `expr`
+Without Aqua's tests, the developers of `PkgA` might not realize that their
+package is essentially unusable with any other package.
```
Comment (NHDaly) [inline docs/src/persistent_tasks.md:47]: 👍

**Review by NHDaly — APPROVED** [review]
❤️ I love it. Thank you!

**Comment by lgoettgens** [comment]
Thanks for the PR. If you and @NHDaly are happy with the changes, there is no further action required from your side. I'll go ahead with updating the changelog and preparing a release with this.
If this PR is still open in a few days, I might have forgotten it. Feel free to ping me then.

**Comment by timholy** [comment]
Thanks for the review and for merging it! And of course especially for your ongoing maintenance, @lgoettgens, of this crucial package!

**Pull request merged**